datajoint/schema.py

```diff
@@ -151,15 +152,26 @@ def __call__(self, cls):
 
         if issubclass(cls, Part):
             raise DataJointError('The schema decorator should not be applied to Part relations')
-        self.process_relation_class(cls, context=self.context)
+        ext = {
+            cls.__name__: cls,
+            'self': cls
+        }
+        self.process_relation_class(cls, context=dict(self.context, **ext))
+
         # Process part relations
-        for part in cls._ordered_class_members:
+        for part in dir(cls):
```

Comment:
Very good point, I have now implemented an extended version of `dir` that respects order of declarations.